### ISSUE TITLE
None finding in report

### ISSUE THREAD

**Issue by zcrt** [minvws/nl-kat-coordination#3759]
**Describe the bug**
Compliance issue: `None`

![image](https://github.com/user-attachments/assets/5287de87-3ac4-4194-b68c-85d1e8b85b04)

**Expected behavior**
No information, or descriptive information

**OpenKAT version**
main 29 okt

**Comment by originalsouth** [comment]
The problem here is that the finding/types are not affirmed yet -- other than improving the default report template for such objects -- we will have to make a trade-off (which can be optionally forwarded to the user): 
* **Scenario 1**: optimize for report generation speed, while keeping the finding un-affirmed and reporting the information is still pending.
* **Scenario 2**: optimize for report completeness, make sure all findings of the report are affirmed. In order achieve this snapshot the report as is and prioritize all affirmation jobs in the scheduler; as there are limited workers the generation of such a report will take much more time is ideally done in an asynchronous way to prevent time-outs).

Thoughts @underdarknl @jpbruinsslot and/or other matters of the report team?

**Comment by jpbruinsslot** [comment]
If we can differentiate from either the ooi or boefje if a scan profile mutations results into an affirmation job then we can give it a higher or the highest priority.

**Comment by originalsouth** [comment]
> If we can differentiate from either the ooi or boefje if a scan profile mutations results into an affirmation job then we can give it a higher or the highest priority.

https://github.com/minvws/nl-kat-coordination/blob/0d1e3cc01d921d346160d2cd6212c8972536169f/octopoes/octopoes/models/ooi/findings.py#L34-L46

As long as `risk_score` and `risk_severity` are unset or set to `0.0` and `pending` respectively, the finding is un-affirmed.

See the history progression of such an object:
```
[
  {
    "contentHash": "0c43cdf9f5f2bae6911dc56b3b843c3327fcb6f7",
    "doc": {
      "KATFindingType/id": "KAT-INVALID-SPF",
      "KATFindingType/primary_key": "KATFindingType|KAT-INVALID-SPF",
      "object_type": "KATFindingType",
      "user_id": null,
      "xt/id": "KATFindingType|KAT-INVALID-SPF"
    },
    "txId": 382,
    "txTime": "2024-10-28T12:40:33Z",
    "validTime": "2024-10-28T12:40:32Z"
  },
  {
    "contentHash": "537edc4c1e937a4ed3c83b2e0e56fe40b6c81a0e",
    "doc": {
      "KATFindingType/id": "KAT-INVALID-SPF",
      "KATFindingType/primary_key": "KATFindingType|KAT-INVALID-SPF",
      "KATFindingType/risk_score": 0.0,
      "KATFindingType/risk_severity": "pending",
      "object_type": "KATFindingType",
      "user_id": null,
      "xt/id": "KATFindingType|KAT-INVALID-SPF"
    },
    "txId": 387,
    "txTime": "2024-10-28T12:40:33Z",
    "validTime": "2024-10-28T12:40:32Z"
  },
  {
    "contentHash": "537edc4c1e937a4ed3c83b2e0e56fe40b6c81a0e",
    "doc": {
      "KATFindingType/id": "KAT-INVALID-SPF",
      "KATFindingType/primary_key": "KATFindingType|KAT-INVALID-SPF",
      "KATFindingType/risk_score": 0.0,
      "KATFindingType/risk_severity": "pending",
      "object_type": "KATFindingType",
      "user_id": null,
      "xt/id": "KATFindingType|KAT-INVALID-SPF"
    },
    "txId": 388,
    "txTime": "2024-10-28T12:40:33Z",
    "validTime": "2024-10-28T12:40:32Z"
  },
  {
    "contentHash": "6a5dfeed7243b042bbbcd18dbcc6e31d5a50558d",
    "doc": {
      "KATFindingType/description": "This SPF record could not be parsed by the internet.nl SPF parser and is therefore deemed invalid.",
      "KATFindingType/id": "KAT-INVALID-SPF",
      "KATFindingType/impact": "E-mail from this domain can potentially be spoofed if DMARC is not (properly) implemented in combination with DKIM and SPF.",
      "KATFindingType/primary_key": "KATFindingType|KAT-INVALID-SPF",
      "KATFindingType/recommendation": "Fix the syntax of the SPF record.",
      "KATFindingType/risk_score": 3.9,
      "KATFindingType/risk_severity": "low",
      "KATFindingType/source": "https://www.cloudflare.com/en-gb/learning/dns/dns-records/dns-spf-record/",
      "object_type": "KATFindingType",
      "user_id": null,
      "xt/id": "KATFindingType|KAT-INVALID-SPF"
    },
    "txId": 400,
    "txTime": "2024-10-28T12:40:44Z",
    "validTime": "2024-10-28T12:40:38Z"
  }
]
```

**Comment by noamblitz** [comment]
I think the problem is that the description is not avaibale yet and therefore `None` is shown. Can't we show the show the unaffirmed finding type id with Pending status when no description is available yet? Seems like a simple solution.

**Comment by originalsouth** [comment]
> I think the problem is that the description is not avaibale yet and therefore `None` is shown. Can't we show the show the unaffirmed finding type id with Pending status when no description is available yet? Seems like a simple solution.

Yes, that has to be done regardless. From @zcrt there is a wish, however, to have all findings in a report affirmed; at least optionally.

**Comment by noamblitz** [comment]
Wait another day :)

**Comment by originalsouth** [comment]
> Wait another day :)

I'm just proposing technical solutions to the request. If that is how we want to solve it, sure...

**Comment by noamblitz** [comment]
Haha no that doesn't seem like the best solution :). But we have this problem everywhere in KAT. So then we should solve it generically, for example we report NO-CERT when no cert run has been done.

**Comment by originalsouth** [comment]
Yeah we have binary findings, they are either/or (not in the Kierkegaardian sense): we should *probably* have no-finding/possible-finding/confirmed-finding.

**Comment by noamblitz** [comment]
Could be indeed. But we're on the edge of a very slippery slide to a very deep rabbit hole haha. Waiting a day, altough not mathematically sound, does solve the problem...

**Comment by originalsouth** [comment]
![image](https://github.com/user-attachments/assets/0b1aac40-cb02-4b44-a67b-29ab05f12d5b)

**Comment by underdarknl** [comment]
> Could be indeed. But we're on the edge of a very slippery slide to a very deep rabbit hole haha. Waiting a day, altough not mathematically sound, does solve the problem...

One of the sub-problems is knowing *when* we are 'done'.
This hinges on a bunch of underlying questions when dealing with reports, such as:
* which OOI's there considered in the queries (including those that did not yield any Objects that ended up in the resulting report)
* Which boefje jobs can be ran on those OOI's
* Which normalizers should have ran on the resulting raw files (only possible to tell once the boefjes have produced raw fiels)
* What is the last run timestamp for each of these jobs
* What is the expected interval for these, and are any 'too old'?
* Did all the bits run?

We could do with an epic for this and start working on these subproblems. Some of which already are listed in our bug-tracker.

**Comment by madelondohmen** [comment]
Wouldn't it be a good idea (to avoid confusion for the user) to add a column "Finding type" to the table, like this:

![afbeelding](https://github.com/user-attachments/assets/b3ad4234-23ff-4141-a130-00178b666faf)

And maybe change "None" to "-" of "Unknown" or something like that?

**Comment by originalsouth** [comment]
Love it, only concern would be that compliance issue can be a long story, I think, so it might get a bit garbled maybe? 

> And maybe change "None" to "-" or "Unknown" or something like that?

Perhaps, "retrieving" so the user know the action is in progress?

**Comment by zcrt** [comment]
Adding the finding type together with (hyperlink to) the information from https://github.com/minvws/nl-kat-coordination/blob/193764c28bf1904be2c7ca4d55e0884f7936b92e/boefjes/boefjes/plugins/kat_kat_finding_types/kat_finding_types.json#L436C1-L442C5 would already be a great improvement. Combined with replacing `None`, by something like `Information not yet available` (potentially hyperlinked to a page on `docs.openkat.nl` explaining what is going on, that would probably be sufficient to solve this issue
